src/Mvc/Mvc.Core/src/ApplicationModels/AuthorizationApplicationModelProvider.cs

```diff
@@ -32,6 +37,13 @@ public void OnProvidersExecuting(ApplicationModelProviderContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            if (_mvcOptions.EnableEndpointRouting)
+            {
+                // When using endpoint routing, the AuthorizationMiddleware does the work that Auth filters would otherwise perform.
```

Comment:
👍

Reply:
If ever there were a place you wanted to confirm that the auth middlewere was added...

Reply:
Right, this could work for MVC controllers, not for pages. It doesn't help with the case also where you put auth middleware somewhere crazy.